docs/ext/thread-pools.md

```diff
@@ -0,0 +1,65 @@
+# Thread Pools
+
+Java's [ThreadPoolExecutor](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html)
+exposes several properties that are useful to monitor to assess the health, performance, and configuration of the pool.
+
+## Getting Started
+
+To report thread pool metrics, one can attach a
+[ThreadPoolMonitor](http://netflix.github.io/spectator/en/latest/javadoc/spectator-api/com/netflix/spectator/api/patterns/ThreadPoolMonitor.html)
+in the following manner:
+
+```java
+import com.netflix.spectator.api.patterns.ThreadPoolMonitor;
+import com.netflix.spectator.api.Spectator;
+
+ThreadPoolMonitor.attach(Spectator.registry(), myThreadPoolExecutor, "my-thread-pool");
+
+```
+
+The provided thread pool will be registered using the provided registry and polled. The metrics will be additionally
+tagged with a key of `id` and the value of the third parameter, if provided, or a default if not.
+
+## Metrics
+
+### threadpool.taskCount
+
+Counter of the total number of tasks that have been scheduled.
+
+**Data Source:** `ThreadPoolExecutor#getTaskCount()`
```

Comment:
Please include the unit for each metric, e.g.: http://netflix.github.io/spectator/en/latest/ext/jvm-gc/#jvmgcpromotionrate